src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -263,14 +266,17 @@ class FunSpec private constructor(builder: Builder) {
 
     fun receiver(receiverType: KClass<*>) = receiver(receiverType.asTypeName())
 
-    fun returns(returnType: TypeName) = apply {
+    @JvmOverloads fun returns(returnType: TypeName, kdoc: CodeBlock? = null) = apply {
       check(!name.isConstructor && !name.isAccessor) { "$name cannot have a return type" }
       this.returnType = returnType
+      kdoc?.let {
+        this.returnKdoc.add("@return %L", it)
```

Comment:
I think we should also be emitting the `\n` at the end of the KDoc. How will the output look if the user doesn't supply the `\n`?

Reply:
Yeah I think it makes sense. It looks pretty weird if the user doesn't add the `\n`. I'll make the changes soon!